drivers/dma/dma_intel_adsp_gpdma.c

```diff
@@ -317,14 +376,25 @@ int intel_adsp_gpdma_get_attribute(const struct device *dev, uint32_t type, uint
 	return 0;
 }
 
+int intel_adsp_gpdma_init(const struct device *dev)
+{
+#if CONFIG_PM_DEVICE
+	pm_device_init_off(dev);
+	return pm_device_runtime_enable(dev);
+#else
+	return intel_adsp_gpdma_power_on(dev);
+#endif
+}
 #ifdef CONFIG_PM_DEVICE
 static int gpdma_pm_action(const struct device *dev, enum pm_device_action action)
 {
 	switch (action) {
-	case PM_DEVICE_ACTION_SUSPEND:
-	case PM_DEVICE_ACTION_RESUME:
 	case PM_DEVICE_ACTION_TURN_ON:
+		return intel_adsp_gpdma_power_on(dev);
 	case PM_DEVICE_ACTION_TURN_OFF:
+		return intel_adsp_gpdma_power_off(dev);
```

Comment:
this only happens if this device is under a domain and the domain is suspended

Reply:
I'm aware of it but this looks like the only right place to do this. I want to turn off device and only PD can trigger TURN_OFF action.

Reply:
@ceolin I decide to move power on and off code to SUSPEND/RESUME actions. It may not fit to the name but it's much better to power on only the device that are in use and not all of them, just because they are in the same power domain.